lib/y2_installbase.pm

```diff
@@ -255,6 +274,50 @@ sub select_all_patterns_by_menu {
     assert_screen 'inst-overview';
 }
 
+=head2 select_not_install_any_pattern 
+
+    select_not_install_any_pattern() 
+
+Being in the "select pattern" screen, performs steps to not install any
+patterns.
+=cut
+
+sub select_not_install_any_pattern {
+    my ($self) = @_;
+
+    # Ensure mouse on certain pattern then right click
+    assert_and_click("minimal-system", button => 'right');
+    assert_screen 'selection-menu';
+    # select action on all patterns
+    wait_screen_change { send_key 'a'; };
+    # confirm do not install
+    assert_and_click 'all-do-not-install';
+    save_screenshot;
+}
+
+=head2 select_current_visible_patterns 
+
+    select_current_visible_patterns()
+
+Being in the "select pattern" screen, performs steps to select visible
+patterns.
+=cut
+
+sub select_current_visible_patterns {
+    my ($self) = @_;
+    my @patterns = split(/,/, get_var('PATTERNS', ''));
+    # remove empty elements
+    @patterns = grep { $_ ne '' } @patterns;
+
+    for my $pattern (@patterns) {
+        # go to the top of the list before looking for the pattern
+        send_key "home";
```

Comment:
I would put this line at the level of test module, besides you don't need to do it for each iteration.

Reply:
When I want to search one pattern, I need put the pattern list part to the first page.
If the pattern cannot be found at first page, I need send down key to move the page to the next.
So for each pattern, I need put the key to 'home' first.

Reply:
When I want to search one pattern, I need put the pattern list part to the first page.
If the pattern cannot be found at first page, I need send down key to move the page to the next.
So for each pattern, I need put the key to 'home' first.

Reply:
putting it out we can do the same from the bottom in another test module.
I still don't get why you need 'down', the pattern should be always found (unselected), because this method only works for visible patterns.